CommMicro.py

```diff
@@ -153,66 +180,63 @@ def getUserVal(self):
+        cmNumStr = cmid.split(':')[2]
 
         # read which rack - 0=A, 1=B
-        rack=self.ui.CavComboBox.currentIndex()
-        if rack==0:
-          delta=1
+        rack = self.ui.CavComboBox.currentIndex()
+        if rack == 0:
+            delta = 1
         else:
-          delta=5
+            delta = 5
 
         # load up cavNumStr ('1234') and cavNumList (['1','2','3','4'])
-        for idx,cb in enumerate(self.checkboxes):
-          if cb.isChecked():
-            cavNumStr += str(idx+delta)
-            cavNumList += str(idx+delta)
+        for idx, cb in enumerate(self.checkboxes):
+            if cb.isChecked():
+                cavNumStr += str(idx + delta)
+                cavNumList += str(idx + delta)
 
         # Make the path name to be nice
-#        LASTPATH=DATA_DIR_PATH+'ACCL_'+liNac+'_'+cmNumStr+cavNumStr[0]+'0'
-        LASTPATH=path.join(DATA_DIR_PATH,'ACCL_'+liNac+'_'+cmNumStr+'00')
+        #        LASTPATH=DATA_DIR_PATH+'ACCL_'+liNac+'_'+cmNumStr+cavNumStr[0]+'0'
+        LASTPATH = path.join(DATA_DIR_PATH, 'ACCL_' + liNac + '_' + cmNumStr + '00')
 
         # get today's date as 2- or 4-char strings
-        year=str(self.startd.year)
-        month='%02d' % self.startd.month
-        day='%02d' % self.startd.day
-        LASTPATH=path.join(LASTPATH,year,month,day)
+        year = str(self.startd.year)
+        month = '%02d' % self.startd.month
+        day = '%02d' % self.startd.day
+        LASTPATH = path.join(LASTPATH, year, month, day)
 
         return liNac, cmNumStr, cavNumStr
 
-# setGOVal is the response to the Get New Measurement button push
-# it takes GUI settings and calls python script to fetch the data
-#  then if Plotting is chosen, it calls getDataBack to make the plot
+    # setGOVal is the response to the Get New Measurement button push
+    # it takes GUI settings and calls python script to fetch the data
+    #  then if Plotting is chosen, it calls getDataBack to make the plot
 
-    def setGOVal(self,tPlot,bPlot):
+    def setGOVal(self, tPlot, bPlot):
         global LASTPATH
-        return_code=2
+        return_code = 2
 
         # reads GUI inputs, fills out LASTPATH, and returns LxB, CMxx, and cav num
-        liNac, cmNumSt, cavNumStr = self.getUserVal()    
-
-        # Get time for measurement from spinbox
-        timMeas = self.ui.TimSpinBox.value()  
+        liNac, cmNumSt, cavNumStr = self.getUserVal()
 
         self.ui.AcqProg.setText("Data acquisition started\n")
         self.ui.AcqProg.repaint()
 
         resScrptSrce = "/usr/local/lcls/package/lcls2_llrf/srf/software/res_ctl/res_data_acq.py"
 
         # made the channel access spec for script call
-        rack=self.ui.CavComboBox.currentIndex()
-        AB='AB'
-        caCmd = "ca://ACCL:"+liNac+":"+str(cmNumSt)+"00:RES"+AB[rack]+":"
+        rack = self.ui.CavComboBox.currentIndex()
+        AB = 'AB'
+        caCmd = "ca://ACCL:" + liNac + ":" + str(cmNumSt) + "00:RES" + AB[rack] + ":"

```

Comment:
The problem is that the combobox doesn't actually have A and B in it, but 'Cavity 1-4' and 'Cavity 5-8'. I think I'll leave it as it is right now since the program works, and in a future re-write with the lcls-tools infrastructure, the problem will 'go away' anyway

Reply:
I don't think that would be super intuitive. Would need at least some extra annotation on the GUI that cavity 1-4  are in rack A and 5-8 are in rack B. As I said, I'll rewrite it with lcls-tools and then the choice of cavity will have the racks solved automatically.